File: components/frontend/src/types/project-settings.ts
export type LLMSettings = {
  model: string;
  temperature: number;
  maxTokens: number;
};

export type ProjectDefaultSettings = {
  llmSettings: LLMSettings;
  defaultTimeout: number;
  allowedWebsiteDomains?: string[];
  maxConcurrentSessions: number;
};

export type ProjectResourceLimits = {
  maxCpuPerSession: string;
  maxMemoryPerSession: string;
  maxStoragePerSession: string;
  diskQuotaGB: number;
};

export type ObjectMeta = {
  name: string;
  namespace: string;
  creationTimestamp: string;
  uid?: string;
};

export type ProjectSettings = {
  projectName: string;
  adminUsers: string[];
  defaultSettings: ProjectDefaultSettings;
  resourceLimits: ProjectResourceLimits;
  metadata: ObjectMeta;
  repos?: ProjectRepo[];
};

export type ProjectSettingsUpdateRequest = {
  projectName: string;
  adminUsers: string[];
  defaultSettings: ProjectDefaultSettings;
  resourceLimits: ProjectResourceLimits;
  repos?: ProjectRepo[];
};

// New types for repository management
export type GroupAccess = {
  groupName: string;
  role: 'admin' | 'edit' | 'view';
};

export type ProjectRepo = {
  name: string;
  url: string;
  defaultBranch?: string;
};

File: components/backend/types/projectsettings.go
package types

// ProjectSettings represents the ProjectSettings CRD spec
type ProjectSettings struct {
	GroupAccess       []GroupAccess `json:"groupAccess,omitempty"`
	RunnerSecretsName string        `json:"runnerSecretsName,omitempty"`
	Repos             []ProjectRepo `json:"repos,omitempty"`
}

// GroupAccess represents RBAC group configuration
type GroupAccess struct {
	GroupName string `json:"groupName" binding:"required"`
	Role      string `json:"role" binding:"required"` // admin, edit, or view
}

// ProjectRepo represents a repository available to this project
type ProjectRepo struct {
	Name          string `json:"name" binding:"required"`
	URL           string `json:"url" binding:"required"`
	DefaultBranch string `json:"defaultBranch,omitempty"`
}
